x-pack/snapshot-tool/build.gradle

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import java.nio.file.Files
+import java.nio.file.Paths
+import org.elasticsearch.gradle.ElasticsearchDistribution
+
+apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.test.fixtures'
+
+dependencies {
+  compile project(":server")
+  compile project(":libs:elasticsearch-cli")
+  compile "com.amazonaws:aws-java-sdk-s3:${versions.aws}"
+  compile "com.amazonaws:aws-java-sdk-core:${versions.aws}"
+  compile "com.amazonaws:jmespath-java:${versions.aws}"
+  compile "org.apache.httpcomponents:httpclient:${versions.httpclient}"
+  compile "org.apache.httpcomponents:httpcore:${versions.httpcore}"
+  compile "commons-logging:commons-logging:${versions.commonslogging}"
+  compile "commons-codec:commons-codec:${versions.commonscodec}"
+  compile "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
+  compile "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
+  compile 'com.fasterxml.jackson.core:jackson-databind:2.8.11.3'
+  compile "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
+  testCompile project(":test:framework")
+  testCompile project(":plugins:repository-s3")
+
+  // HACK: javax.xml.bind was removed from default modules in java 9, so we pull the api in here,
+  // and whitelist this hack in JarHell
+  compile 'javax.xml.bind:jaxb-api:2.2.2'
```

Comment:
Could this be the reason it fails on my local CI server with the following error?
```
[2019-07-19T19:08:37.414Z] > Task :x-pack:snapshot-tool:smokeTest FAILED
[2019-07-19T19:08:37.414Z] Error: A JNI error has occurred, please check your installation and try again
[2019-07-19T19:08:37.414Z] Exception in thread "main" java.lang.UnsupportedClassVersionError: org/elasticsearch/snapshots/SnapshotToolCli has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
[2019-07-19T19:08:37.414Z] 	at java.lang.ClassLoader.defineClass1(Native Method)
[2019-07-19T19:08:37.414Z] 	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
[2019-07-19T19:08:37.414Z] 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
[2019-07-19T19:08:37.414Z] 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
[2019-07-19T19:08:37.414Z] 	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
[2019-07-19T19:08:37.414Z] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
[2019-07-19T19:08:37.414Z] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
[2019-07-19T19:08:37.414Z] 	at java.security.AccessController.doPrivileged(Native Method)
[2019-07-19T19:08:37.414Z] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
[2019-07-19T19:08:37.414Z] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[2019-07-19T19:08:37.414Z] 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
[2019-07-19T19:08:37.414Z] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[2019-07-19T19:08:37.414Z] 	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
[2019-07-19T19:08:37.414Z] 
[2019-07-19T19:08:37.414Z] FAILURE: Build failed with an exception.
[2019-07-19T19:08:37.414Z] 
[2019-07-19T19:08:37.414Z] * What went wrong:
[2019-07-19T19:08:37.414Z] Execution failed for task ':x-pack:snapshot-tool:smokeTest'.
[2019-07-19T19:08:37.414Z] > Process 'command '/var/jenkins_home/jobs/elasticsearch-pipeline/workspace/elasticsearch/x-pack/snapshot-tool/build/snapshot-tool-8.0.0-SNAPSHOT/bin/elasticsearch-snapshot'' finished with non-zero exit value 1
[2019-07-19T19:08:37.414Z] 
[2019-07-19T19:08:37.414Z] * Try:
[2019-07-19T19:08:37.414Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[2019-07-19T19:08:37.414Z] 
[2019-07-19T19:08:37.414Z] * Get more help at https://help.gradle.org
[2019-07-19T19:08:37.414Z] 
[2019-07-19T19:08:37.414Z] BUILD FAILED in 15m 21s
```

Reply:
@imotov I think the problem here might rather be the somewhat archaic script that starts this tool from the CLI.

See the third line in your snippet:
```
] Exception in thread "main" java.lang.UnsupportedClassVersionError: org/elasticsearch/snapshots/SnapshotToolCli has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

Could it be that you have `java` bound to some Java 8 version in your environment and if we compile things for Java 11 and simply use `java` in the script outright now it fails for that reason?

Reply:
@original-brownbear Right I misread it in the reverse order. I am using our standard elasticsearch-ci setup to run it. It might have Java 8 bound to Java indeed.

Reply:
@andrershov I think this is something we should fix. This is going to be a pain to maintain in CI otherwise.

Reply:
@andrershov @original-brownbear  Is there a workaround for this?

Reply:
@imotov I don't think so (other than not binding java 8 to `java`). But if this is a blocker I can look into this tomorrow?

Reply:
I cannot use elasticsearch-ci because of this. So, we need to fix that here or in the elasticsearch-ci script.

Reply:
@imotov understood, looking into this now then.

Reply:
This should do it https://github.com/elastic/elasticsearch/pull/44919